tests/unittests/algo/test_evolution_es.py

```diff
@@ -237,3 +182,83 @@ def test_register(self, evolution, bracket, rung_0, rung_1):
         assert len(bracket.rungs[0])
         assert point_hash in bracket.rungs[0]['results']
         assert (0.0, point) == bracket.rungs[0]['results'][point_hash]
+
+
+class TestBracketEVES():
+    """Tests for `BracketEVES` class.."""
+
+    def test_get_teams(self, bracket, rung_3):
+        """Test that correct team is promoted."""
+        bracket.rungs[0] = rung_3
+        rung, population_range, red_team, blue_team = bracket._get_teams(0)
+        assert len(list(rung.values())) == 4
+        assert bracket.search_space_remove_fidelity == [1, 2]
+        assert population_range == 4
+        assert set(red_team).union(set(blue_team)) == {0, 1, 2, 3}
+        assert set(red_team).intersection(set(blue_team)) == set()
+
+    def test_get_mutated_population(self, bracket, rung_3):
+        """Verify mutated candidates is generated correctly."""
+        red_team = [0, 2]
+        blue_team = [1, 3]
+        for i in range(4):
+            for j in [1, 2]:
+                bracket.eves.population[j][i] = list(rung_3["results"].values())[i][1][j]
+            bracket.eves.performance[i] = list(rung_3["results"].values())[i][0]
+
+        org_data = np.stack((list(bracket.eves.population.values())[0],
+                             list(bracket.eves.population.values())[1]), axis=0).T
+
+        org_data = copy.deepcopy(org_data)
+
+        bracket._get_mutated_population(red_team, blue_team)
+
+        mutated_data = np.stack((list(bracket.eves.population.values())[0],
+                                 list(bracket.eves.population.values())[1]), axis=0).T
+
+        assert org_data.shape == mutated_data.shape
+        assert (mutated_data[0] == org_data[0]).all()
```

Comment:
Could you please document the rational here for the tests. It's hard to infer quickly. Thanks!

Reply:
Done. For my test case: Winner team will be [0, 2], so [0, 2] will be remained, while [1, 3] will be mutated. In addition, for each individual, mutation occurs in only one dimension chosen from two. So for example, if lr is mutated, weight decay will be copied from individual 0/2 directly, and vice versa. I added this comments in code.